conda/base/context.py

```diff
@@ -964,7 +976,12 @@ def channels(self):
                     "--override-channels."
                 )
             else:
-                return tuple(IndexedSet((*local_add, *self._argparse_args["channel"])))
+                channels = tuple(
+                    IndexedSet((*local_add, *self._argparse_args["channel"]))
+                )
```

Comment:
can we also add the `tuple(IndexedSet)` to `check_allowlist` as well? E.g.:

```python
return check_allowlist((*local_add, *self._argparse_args["channel"]))
```

Reply:
I would rather not do this because it would mean changing the return type of that function and we would have to set up a deprecation warning first.

Reply:
now that we have a new function (`check_channel_allowlist`) and are deprecating the old one (`check_allowlist`) can we shove this into the new function?